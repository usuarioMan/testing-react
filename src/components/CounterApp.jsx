import { useState } from "react";


export const CounterApp = ({value}) => {
    const [cuenta, setCuenta] = useState(value);
 return (
    <>
    <div>
    <h1>CounterApp</h1>
    <div>{cuenta}</div>
    <button onClick={() => {
      setCuenta(cuenta-1)
    }}>-1</button>
    <button onClick={() => {
      setCuenta(cuenta+1)
    }}>+1</button>
    <button onClick={() => {
      setCuenta(value)
    }}>Reset</button>
    </div>
    </>
  )
}
