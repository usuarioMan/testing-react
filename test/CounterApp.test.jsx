import { CounterApp } from "../src/components/CounterApp";
import { render, screen} from "@testing-library/react";


describe('Test en CounterApp', () => {
    
    test('Esta prueba debe coincidir con el snapshot', () => {
    const {container} = render(<CounterApp value={420}/>)
    expect(container).toMatchSnapshot()
    
   })
   
   test('Esta prueba debe iniciar el valor inicial 420', () => {
    render(<CounterApp value={420}/>)
    expect(screen.getByText(420)).toBeTruthy()
    
   })

 })