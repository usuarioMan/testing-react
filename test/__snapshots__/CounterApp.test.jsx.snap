// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en CounterApp Esta prueba debe coincidir con el snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <div>
      420
    </div>
    <button>
      -1
    </button>
    <button>
      +1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
